dev-packages/application-manager/src/application-package-manager.ts

```diff
@@ -117,10 +117,33 @@ export class ApplicationPackageManager {
     start(args: string[] = []): cp.ChildProcess {
         if (this.pck.isElectron()) {
             return this.startElectron(args);
+        } else if (this.pck.isBrowserOnly()) {
+            return this.startBrowserOnly(args);
         }
         return this.startBrowser(args);
     }
 
+    startBrowserOnly(args: string[]): cp.ChildProcess {
+        const { command, mainArgs, options } = this.adjustBrowserOnlyArgs(args);
+        return this.__process.spawnBin(command, mainArgs, options);
+    }
+
+    adjustBrowserOnlyArgs(args: string[]): Readonly<{ command: string, mainArgs: string[]; options: cp.SpawnOptions }> {
+        let { mainArgs, options } = this.adjustArgs(args);
+
+        mainArgs = ['serve', ...mainArgs];
+
+        const portIndex = mainArgs.findIndex(v => v.startsWith('--port'));
+        if (portIndex === -1) {
+            mainArgs.push('--port=3000');
+        }
+        const configIndex = mainArgs.findIndex(v => v.startsWith('--config') || v.startsWith('-c'));
+        if (configIndex === -1) {
+            mainArgs.push('--config=webpack.config.js');
+        }
+        return { command: 'webpack', mainArgs, options };
```

Comment:
Discussion: We don't use `webpack serve` for any other configuration. I'm not totally against it, I just didn't expect to see it here. I would've expected a lighter solution via something like `http-server`. Especially since we have `watch` and `build` scripts in our `browser-only` app, which shouldn't be needed with `webpack serve`, right?

Note that I've noticed this in particular, because `webpack serve` returns `Invalid Host header` when trying to access the website using Gitpod for some reason.

Reply:
You are right. `http-server` is much more lightweight and should serve fine here!